pyfstat/core.py

```diff
@@ -661,16 +669,20 @@ def init_computefstatistic(self):
 
         # always initialise the twoFX array,
         # but only actually compute it if requested
+        self.twoF = 0
         self.twoFX = np.zeros(lalpulsar.PULSAR_MAX_DETECTORS)
         self.singleFstats = self.singleFstats or self.BSGL  # BSGL implies twoFX
         if self.singleFstats:
             self.whatToCompute += lalpulsar.FSTATQ_2F_PER_DET
 
         if self.BSGL:
+            self.log10BSGL = np.nan
             if len(self.detector_names) < 2:
                 raise ValueError("Can't use BSGL with single detectors data")
             else:
                 logging.info("Initialising BSGL")
+            if getattr(self, "BtSG", False):
```

Comment:
Can this be moved earlier during initialization?

Reply:
probably, will include in a cleanup commit